### PR TITLE
Add grpc-tags-bin propagation helpers.

### PR DESCRIPTION
**Pull request by g-easy**

### CODE REVIEW

File: opencensus/tags/internal/grpc_tags_bin.cc [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+}
+
+// Parses a variable-length encoded integer from the input. Returns false on
+// failure. Returns true and consumes the bytes from the input, on success.
+bool ParseVarint(absl::string_view* input, int* out) {
+  absl::string_view s = *input;
+  int i = 0;
+  uint8_t c;
+  do {
+    if (s.length() < 1) {
```
[the remark this answers is not in the document]
Comment (g-easy) [reply opencensus/tags/internal/grpc_tags_bin.cc]: Done.

File: opencensus/tags/internal/grpc_tags_bin.cc [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    return false;  // Too long.
+  }
+  if (header[0] != kVersionId) {
+    return false;  // Wrong version.
+  }
+  header = header.substr(1);
+  while (header.length() > 0) {
+    // Parse tag field id.
+    if (header.length() < 1) {
+      return false;  // Too short.
```
[the remark this answers is not in the document]
Comment (g-easy) [reply opencensus/tags/internal/grpc_tags_bin.cc]: Removed.

File: opencensus/tags/internal/grpc_tags_bin_test.cc [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    const std::string k = absl::StrCat("key_", key_n++);
+    tags.emplace_back(TagKey::Register(k), std::string('A', kValLen));
+    out_len += 1 + 1 + k.size() + 1 + kValLen;
+  }
+  TagMap m(std::move(tags));
+  EXPECT_EQ("", ToGrpcTagsBinHeader(m))
+      << "Serialization failed due to value being too long.";
+}
+
+// TODO: empty value?
```
[the remark this answers is not in the document]
Comment (g-easy) [reply opencensus/tags/internal/grpc_tags_bin_test.cc]: What do you mean?

File: opencensus/tags/internal/grpc_tags_bin_benchmark.cc [hunk 4]
```diff
@@ -0,0 +1 @@
+// TODO
```
[the remark this answers is not in the document]
Comment (g-easy) [reply opencensus/tags/internal/grpc_tags_bin_benchmark.cc]: Added benchmark.

**Comment by g-easy** [comment]
```
Benchmark                         Time             CPU   Iterations
-------------------------------------------------------------------
BM_FromGrpcTagsBinHeader        147 ns          147 ns      3875450
BM_ToGrpcTagsBinHeader         44.7 ns         44.7 ns     15636119
```

File: opencensus/tags/internal/grpc_tags_bin_test.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply opencensus/tags/internal/grpc_tags_bin_test.cc]: Please explain the todo better

[the remark this answers is not in the document]
Comment (g-easy) [reply opencensus/tags/internal/grpc_tags_bin_test.cc]: What happens if the key and/or value is the empty string? Where should that be enforced? Also, we don't (in C++) enforce that the value is valid UTF-8 at any point.

(I can update the code to spell this out or drop the TODOs, but more importantly, how should we actually handle these cases?)

[the remark this answers is not in the document]
Comment (bogdandrutu) [reply opencensus/tags/internal/grpc_tags_bin_test.cc]: If the key is empty drop the entry. Empty value I think it is ok.

[the remark this answers is not in the document]
Comment (g-easy) [reply opencensus/tags/internal/grpc_tags_bin_test.cc]: Done + added test.

**Pull request merged**